spring-xd-yarn/spring-xd-yarn-client/src/main/java/org/springframework/xd/yarn/XdYarnAdminInfoCommand.java

```diff
@@ -0,0 +1,126 @@
+import org.springframework.core.io.FileSystemResource;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * A custom command for listing the http ports used by admin servers.
+ *
+ * @author Thomas Risberg
+ */
+public class XdYarnAdminInfoCommand extends AbstractCommand {
+
+	private static final String XD_YARN_ADMIN_INFO_COMMAND_NAME = "admininfo";
+	private static final String XD_YARN_ADMIN_INFO_COMMAND_DESC = "List http host and port info for the admin servers";
+
+	public static final String ZK_NAMESPACE = "zk.namespace";
+	public static final String ZK_CLIENT_CONNECT = "zk.client.connect";
+
+	public static final String XD_ADMINS = "admins";
+
+	private String zkNamespace = "xd";
+	private String zkClientConnect = "localhost:2181";
+
+	private ZooKeeper zk;
+
+	/**
+	 * Call super constructor for new {@link org.springframework.boot.cli.command.AbstractCommand} instance.
+	 */
+	public XdYarnAdminInfoCommand() {
+		super(XD_YARN_ADMIN_INFO_COMMAND_NAME, XD_YARN_ADMIN_INFO_COMMAND_DESC);
+	}
+
+	@Override
+	public ExitStatus run(String... args) throws InterruptedException {
+		try {
+			init();
+			Log.info("Admins: " + getAdmins());
+		}
+		catch (IOException e) {
+			Log.error("IOException accessing admin info: " + e.getMessage());
+		}
+		catch (KeeperException e) {
+			Log.error("KeeperException accessing admin info: " + e.getMessage());
+		}
+		finally {
+			close();
+		}
+		return ExitStatus.OK;
+	}
+
+	private void init() throws IOException {
+		YamlPropertiesFactoryBean ypfb = new YamlPropertiesFactoryBean();
+		ypfb.setResources(new FileSystemResource(System.getProperty("spring.config.location")));
+		Properties conf = ypfb.getObject();
+		if (conf.containsKey(ZK_NAMESPACE)) {
+			zkNamespace = (String) conf.get(ZK_NAMESPACE);
+		}
+		if (conf.containsKey(ZK_CLIENT_CONNECT)) {
+			zkClientConnect = (String) conf.get(ZK_CLIENT_CONNECT);
+		}
+		zk = new ZooKeeper(zkClientConnect, 5000,
+				new Watcher() {
+					@Override
+					public void process(WatchedEvent event) {
+					}
+				});
+	}
+
+	private void close() throws InterruptedException {
+		if (zk != null) {
+			zk.close();
+		}
+	}
+
+	private List<String> getAdmins() throws KeeperException {
+		String adminsPath = "/" + zkNamespace + "/" + XD_ADMINS;
+		List<String> results = new ArrayList<String>();
+		try {
+			Stat stat = zk.exists(adminsPath, false);
+			if (stat != null) {
+				List<String> admins = zk.getChildren(adminsPath, false);
+				for (String admin : admins) {
+					Stat dataStat = new Stat();
+					byte[] data = zk.getData(adminsPath + "/" + admin, false, dataStat);
+					try {
+						HashMap<String,Object> dataValues =
+								new ObjectMapper().readValue(new String(data), HashMap.class);
+						results.add("http://" + dataValues.get("host") + ":" + dataValues.get("port"));
+					} catch (IOException e) {}
+				}
+			}
+		} catch (InterruptedException ignore) {}
```

Comment:
The method calling `getAdmins` throws `InterruptedException` - so why bother catching it here? I would just declare that this method throws it and let it through if it happens.